language/moe/evaluate_mbxp.py

```diff
@@ -0,0 +1,208 @@
+import timeit
+
+import pandas as pd
+from tqdm import tqdm
+
+from mxeval.execution import check_correctness as check_correctness_python
+from mxeval.execution import (
+    check_correctness_cpp,
+    check_correctness_csharp,
+    check_correctness_go,
+    check_correctness_java,
+    check_correctness_javascript,
+    check_correctness_kotlin,
+    check_correctness_perl,
+    check_correctness_php,
+    check_correctness_ruby,
+    check_correctness_scala,
+    check_correctness_swift,
+    check_correctness_typescript,
+)
+
+
+def postprocess_golang(code: str) -> str:
+    multi_line_imports = re.compile(r"^import \(\n(.+)((?:\n.+)+)\n\)", re.MULTILINE)
+    line_imports = re.compile(r"^import \".*\"")
+    func_main = re.compile(r"^func main.*^}", re.MULTILINE | re.DOTALL)
+    
+    code = code.replace("package main", "") # Remove package main
+    code = multi_line_imports.sub("", code)
+    code = line_imports.sub("", code)
+    code = func_main.sub("", code)
+
+    return code
+
+def postprocess_scala(code: str) -> str:
+    code = code.replace("object Main extends App {", "")
+    code = "".join(code.splitlines(True)[:-1])
+    return code
+
+def postprocess_python(code: str) -> str:
+    return code.lstrip()
+
+def worker(inp_queue, out_queue):
+    while True:
+        try:
+            problem = inp_queue.get(timeout=5)
+        except queue.Empty:
+            break
+
+        key = f"{problem['lang']}_{problem['entry_point']}"
+        checker = eval(f"check_correctness_{problem['lang']}")
+
+        problem["task_id"] = key
+        problem["test"] = problem["test_code"]
+
+        solution = problem["response"]
+
+        try:
+            solution = solution[:solution.index("```")]
+        except ValueError:
+            #Happens when a code block isn't closed properly
+            pass
+
+        if problem["lang"] == "go":
+            solution = postprocess_golang(solution)
+        elif problem["lang"] == "python":
+            solution = postprocess_python(solution)
+        elif problem["lang"] == "scala":
+            solution = postprocess_scala(solution)
+
+        # Mixtral likes escaping underscores for some reason, so let's remove these
+        solution = solution.replace("\_", "_")
+        
+        # The evaluation script evaluates `code = prompt + solution + tests`
+        # But Mixtral regenerates the prompt in its output, so we should remove this
+        problem["prompt"] = ""
+
+        result = checker(problem, solution, timeout=20.0)
+        out_queue.put((key, problem["lang"], result["passed"], result["result"], problem["response"]))
+
+def convert_pickle(df: pd.DataFrame):
+    problems = []
+    for _, row in df.iterrows():
+        lang, entry_point = row["id"].split("_", 1)
+        problems.append({
+            "lang": lang,
+            "prompt": row["input"],
+            "test_code": row["gt_output"],
+            "entry_point": entry_point,
+            "response": row["ref_output"]
```

Comment:
is this code used? why ref_output?

Reply:
This code is not being used. I took the Nvidia script as a starting point and I didn't remove this. Will remove it